engine/access/rpc/backend/backend_scripts.go

```diff
@@ -128,12 +120,46 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 	// *DO NOT* use this hash for any protocol-related or cryptographic functions.
 	insecureScriptHash := md5.Sum(script) //nolint:gosec
 
-	// try each of the execution nodes found
+	// try execution on Archive nodes first
+	if len(b.archiveAddressList) > 0 {
+		startTime := time.Now()
+		for _, rnAddr := range b.archiveAddressList {
+			result, err := b.tryExecuteScriptOnArchiveNode(ctx, rnAddr, blockID, script, arguments)
+			if err == nil {
+				// log execution time
+				b.metrics.ScriptExecuted(
+					time.Since(startTime),
+					len(script),
+				)
+				return result, nil
+			} else {
+				errCode := status.Code(err)
+				switch errCode {
+				case codes.InvalidArgument:
+					// failure due to cadence script, no need to query further
+					b.log.Debug().Err(err).
+						Str("script_executor_addr", rnAddr).
+						Hex("block_id", blockID[:]).
+						Hex("script_hash", insecureScriptHash[:]).
+						Str("script", string(script)).
+						Msg("script failed to execute on the execution node")
+					return nil, err
+				case codes.NotFound:
+					// failures due to unavailable blocks are explicitly marked Not found
+					b.metrics.ScriptExecutionErrorOnArchiveNode()
+					b.log.Error().Err(err).Msg("script execution failed for archive node internal reasons")
```

Comment:
the error is `NotFound`
```suggestion
					b.log.Error().Err(err).Msg("script execution failed for archive node")
```